raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -360,26 +359,30 @@ private void maybeBecomeLeader(CandidateState state) throws IOException {
         }
     }
 
-    private void onBecomeCandidate(CandidateState state) throws IOException {
+    private void onBecomeCandidate(CandidateState state, int retries) throws IOException {
         maybeBecomeLeader(state);
         resetConnections();
 
         kafkaRaftMetrics.updateElectionStartMs(time.milliseconds());
```

Comment:
Above we have `maybeBecomeLeader`, which I think handles the case when there is just one voter. It seems like we might want to check that we are still a candidate before updating this metric and resetting the timer.

Reply:
Yup, I've done that part but did not pushed the latest commits to this PR.